.github/workflows/cd.yml

```
@@ -0,0 +1,62 @@
+name: CD
+
+on: [push, pull_request]
+
+jobs:
+  pypi:
+    name: build and deploy to PyPI
+    if: github.repository == 'ACCESS-NRI/access-py-telemetry' && github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+    runs-on: "ubuntu-latest"
+    permissions:
+      id-token: write
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4.2.2
+
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
```

Although not super critical, you can cache the installation of later python packages by adding the `cache: pip` argument to the `setup-python` action, like so:
```suggestion
      - name: Set up Python 3.11
        uses: actions/setup-python@v5
        with:
          python-version: 3.11
          cache: pip
```